torch/ao/quantization/qconfig.py

```diff
@@ -80,64 +80,38 @@ def __new__(cls, activation, weight):
 Default qconfig configuration for per channel weight quantization.
 """
 
-class QConfigDynamic(namedtuple('QConfigDynamic', ['activation', 'weight'])):
-    """
-    Describes how to dynamically quantize a layer or a part of the network by providing
-    settings (observer classes) for weights.
-
-    It's like QConfig, but for dynamic quantization.
-
-    Note that QConfigDynamic needs to contain observer **classes** (like MinMaxObserver) or a callable that returns
-    instances on invocation, not the concrete observer instances themselves.
-    Quantization function will instantiate observers multiple times for each of the layers.
-
-    Observer classes have usually reasonable default arguments, but they can be overwritten with `with_args`
-    method (that behaves like functools.partial)::
-
-      my_qconfig = QConfigDynamic(weight=default_observer.with_args(dtype=torch.qint8))
-
-    """
-    def __new__(cls, activation=torch.nn.Identity, weight=torch.nn.Identity):
-        # catch common mistakes
-        if isinstance(weight, nn.Module):
-            raise ValueError("QConfigDynamic received observer instance, please pass observer class instead. " +
-                             "Use MyObserver.with_args(x=1) to override arguments to constructor if needed")
-        return super(QConfigDynamic, cls).__new__(cls, activation, weight)
-
-default_dynamic_qconfig = QConfigDynamic(activation=default_dynamic_quant_observer,
-                                         weight=default_weight_observer)
+default_dynamic_qconfig = QConfig(activation=default_dynamic_quant_observer,
+                                  weight=default_weight_observer)
```

Comment:
Should we have a warning first in case of OSS users who may still be using `QConfigDynamic` in their code? Deleting it would cause an error in the user code.

Reply:
do you mean we first put up a warning and then delete it in a future release?

Reply:
sure, will update the PR

Reply:
updated